### ISSUE TITLE
include_role task with 'static: no' adds an IncludeRole object inside the result._result callback dictionary

### ISSUE THREAD

**Issue by dmsimard** [ansible/ansible#30385]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
include_role

##### ANSIBLE VERSION
```
ansible 2.3.2.0
```

##### CONFIGURATION
Not relevant

##### OS / ENVIRONMENT
Not relevant

##### SUMMARY
The ``include_role`` task, when used with ``static: no``, adds a ``IncludeRole`` object inside the ``result._result`` dictionary in callback hooks (such as ``v2_runner_on_ok``).
This is inconsistent as this key (which is an object) is not present when running ``ìnclude_role`` without the ``static`` parameter and this key is not present either for the ``include`` task.

This behavior was reproduced on Ansible devel (2.5).

This is particularly problematic because the IncludeRole object ends up causing JSON serialization issues which is not triggered by anything else so far as I can tell.

##### STEPS TO REPRODUCE
See reproducer config/playbook/callback here: https://github.com/dmsimard/ansible_includerole_issue

##### EXPECTED RESULTS
The IncludeRole object probably don't belong there, or it would need to be serializable, not a literal class instance.

##### ACTUAL RESULTS
The IncludeRole object can crash things:

```
$ ansible-playbook playbook.yml 
 [WARNING]: Host file not found: /etc/ansible/hosts

 [WARNING]: provided hosts list is empty, only localhost is available


PLAY [Run reproducer without static] **********************************************************************************************************************************************************

TASK [reproducer : debug] *********************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "noop"
}
########################################
{'msg': u'noop', '_ansible_verbose_always': True, '_ansible_no_log': False}
########################################
{u'msg': u'noop'}
########################################

PLAY [Run reproducer with static] *************************************************************************************************************************************************************

TASK [include_role] ***************************************************************************************************************************************************************************
########################################
{'include_variables': {u'name': u'reproducer'}, 'include_role': TASK: include_role, 'changed': False}
 [WARNING]: Failure using method (v2_runner_on_ok) in callback plugin
(<ansible.plugins.callback./home/dmsimard/dev/ansible_includerole_issue/library/plugins/callback/reproducer.CallbackModule object at 0x7f84ec98e190>): TASK: include_role is not JSON
serializable


TASK [reproducer : debug] *********************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "noop"
}
########################################
{'msg': u'noop', '_ansible_verbose_always': True, '_ansible_no_log': False}
########################################
{u'msg': u'noop'}
########################################

PLAY RECAP ************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0
```

**Comment by jborean93** [comment]
cc @bcoca @jimi-c

**Comment by bcoca** [comment]
this is expected, as instead of importing (static=yes) it becomes 'the task' .. include_role is at this point a Task object and that is what we normally pass.

**Comment by bcoca** [comment]
@dmsimard  so I tried to fix serialization issues with task_include in  #28077, please test and let me know if it fixes the basic problem for you.

**Comment by jimi-c** [comment]
This should now be resolved upstream and will be included in 2.5 and 2.4.3.

Thanks!

**Comment by dmsimard** [comment]
For the record, fix in question was https://github.com/ansible/ansible/commit/c30ee42fe1f0a9666a90f4d63121780f2a186c54
